fastlane/SnapshotHelper.swift

```diff
@@ -96,78 +122,124 @@ open class Snapshot: NSObject {
-
-        while (0..<query.count).map({ query.element(boundBy: $0) }).contains(where: { $0.isLoadingIndicator }) {
-            sleep(1)
-            print("Waiting for loading indicator to disappear...")
-        }
+        let networkLoadingIndicator = XCUIApplication().otherElements.deviceStatusBars.networkLoadingIndicators.element
+        let networkLoadingIndicatorDisappeared = XCTNSPredicateExpectation(predicate: NSPredicate(format: "exists == false"), object: networkLoadingIndicator)
+        let _ = XCTWaiter.wait(for: [networkLoadingIndicatorDisappeared], timeout: timeout)
     }
 
-    class func pathPrefix() -> URL? {
+    class func pathPrefix() throws -> URL? {
         let homeDir: URL
         // on OSX config is stored in /Users/<username>/Library
         // and on iOS/tvOS/WatchOS it's in simulator's home dir
         #if os(OSX)
             guard let user = ProcessInfo().environment["USER"] else {
-                print("Couldn't find Snapshot configuration files - can't detect current user ")
-                return nil
+                throw SnapshotError.cannotDetectUser
             }
 
             guard let usersDir =  FileManager.default.urls(for: .userDirectory, in: .localDomainMask).first else {
-                print("Couldn't find Snapshot configuration files - can't detect `Users` dir")
-                return nil
+                throw SnapshotError.cannotFindHomeDirectory
             }
 
             homeDir = usersDir.appendingPathComponent(user)
         #else
             guard let simulatorHostHome = ProcessInfo().environment["SIMULATOR_HOST_HOME"] else {
-                print("Couldn't find simulator home location. Please, check SIMULATOR_HOST_HOME env variable.")
-                return nil
+                throw SnapshotError.cannotFindSimulatorHomeDirectory
             }
             guard let homeDirUrl = URL(string: simulatorHostHome) else {
-                print("Can't prepare environment. Simulator home location is inaccessible. Does \(simulatorHostHome) exist?")
-                return nil
+                throw SnapshotError.cannotAccessSimulatorHomeDirectory(simulatorHostHome)
             }
             homeDir = URL(fileURLWithPath: homeDirUrl.path)
         #endif
         return homeDir.appendingPathComponent("Library/Caches/tools.fastlane")
     }
 }
 
-extension XCUIElement {
-    var isLoadingIndicator: Bool {
-        let whiteListedLoaders = ["GeofenceLocationTrackingOn", "StandardLocationTrackingOn"]
-        if whiteListedLoaders.contains(self.identifier) {
-            return false
+private extension XCUIElementAttributes {
+    var isNetworkLoadingIndicator: Bool {
+        if hasWhiteListedIdentifier { return false }
+
+        let hasOldLoadingIndicatorSize = frame.size == CGSize(width: 10, height: 20)
+        let hasNewLoadingIndicatorSize = frame.size.width.isBetween(46, and: 47) && frame.size.height.isBetween(2, and: 3)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/cc0d6ef80a88a1c59f1d6996e7e1634a38d73d01/fastlane/SnapshotHelper.swift#L191"></span>Line should be 120 characters or less: currently 122 characters</td>
    </tr>
  </tbody>
</table>